include/STREAM/Equations/ConfinementTime.hpp

```diff
@@ -25,15 +24,51 @@ namespace STREAM{
                         EllipticalRadialGridGenerator *radials;
 						DREAM::IonHandler *ions;
                         
-                        len_t id_Tcold, id_Wi, id_Ni; 
+                        len_t id_Tcold, id_Wi, id_Ni, id_Ip, id_Ncold, id_Efield; 
                         
                         ConnectionLength *CL;
                         
                         len_t D_index;
+                        
+                        static constexpr real_t ConversionFactor = 1e-6;
+                        static constexpr real_t nullset = 0.0;
+                        static constexpr real_t Conv_n20 = 1e-20;
+                        
+                        bool mixedConfLaw;
+                        
+                        std::function<real_t(real_t,real_t)> smoothlessFunction;
+                        
+                        EquationSystem *eqsys;
+                        
+                        OptionConstants::Conf_Time_type type;
+                        OptionConstants::Conf_Time_smoothless smoothless;
+                        
+                        real_t GetOhmicPower(len_t ir) const; // Computes the ohmic power
+                        
+                        real_t Bohm_ConfinementTime(len_t ir) const; // Computes the confinement time from Bohm
+                                                
+						real_t INTOR_ConfinementTime(len_t ir) const;
+						
+						real_t ITER89_OL_ConfinementTime(len_t ir) const;
+						
+						real_t OS_OL_ConfinementTime(len_t ir) const;
+						
+						real_t RL_OL_ConfinementTime(len_t ir) const;
+						
+						real_t Goldstone_scaling(PowerList const&, len_t ir, real_t const& ConvUnit = 1e-19) const;
+						
+						const PowerList GetCoeff() const;
+						const PowerList GetCoeff(OptionConstants::Conf_Time_type _type) const;
+						
+						const real_t EvaluatePerpendicularConfinementTimeType(len_t ir) const;
 
+                        std::function<real_t(real_t, real_t)> setSmoothless();
```

Comment:
smooth? smoothless? 
comments required here, not self explanatory

Reply:
This has been removed